File: laravel/Negozio/app/Http/Controllers/StoreController.php
<?php

namespace App\Http\Controllers;

use App\Models\Store;
use Illuminate\Http\Request;

class StoreController extends Controller
{
    /**
     * Display a listing of the resource.
     */
    public function index()
    {
        $stores= Store::all();
        return view("stores.index",compact('stores'));
    }

    /**
     * Show the form for creating a new resource.
     */
    public function create()
    {
        return view("stores.create");
    }

    /**
     * Store a newly created resource in storage.
     */
    public function store(Request $request)
    {
        Store::create($request->all());
        return redirect('/stores');
    }

    /**
     * Display the specified resource.
     */
    public function show(Store $store)
    {
        return view('stores.show',compact('store'));
    }

    /**
     * Show the form for editing the specified resource.
     */
    public function edit(Store $store)
    {
        return view('stores.edit',compact('store'));
    }

    /**
     * Update the specified resource in storage.
     */
    public function update(Request $request, Store $store)
    {
        $store->update($request->all());
        return redirect('/stores');
    }

    /**
     * Remove the specified resource from storage.
     */
    public function destroy(Store $store)
    {
        $store->delete();
        return redirect('/stores');

    }
}

File: javaSpring/Negozio/src/main/java/com/example/Negozio/controllers/StoreController.java
package com.example.Negozio.controllers;

import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;

import com.example.Negozio.models.Store;
import com.example.Negozio.repositories.StoreRepository;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;



@Controller
public class StoreController {
    
    StoreRepository repo;

    public StoreController(StoreRepository repo){
        this.repo=repo;
    }

    @GetMapping("/")
    public String home() {
        return "/index";
    }
    

    @GetMapping("/stores")
    public String index(Model model) {
        model.addAttribute("stores", repo.findAll());
        return "/stores/index";
    }

    @GetMapping("/stores/create")
    public String create() {
        return  "/stores/create";
    }

    @PostMapping("/stores/store")
    public String store(Store store) {
        repo.save(store);
        return "redirect:/stores";
    }  

    @GetMapping("/stores/show")
    public String show(Long id,Model model) {
        model.addAttribute("store", repo.findById(id).orElse(null));
        return "/stores/show";
    }

    @GetMapping("/stores/edit")
    public String edit(Long id,Model model) {
        model.addAttribute("store", repo.findById(id).orElse(null));
        return "/stores/edit";
    }

    @PostMapping("/stores/update")
    public String update(Store store) {
        repo.save(store);
        return "redirect:/stores";
    }

    @PostMapping("/stores/delete")
    public String delete(Long id) {
        repo.deleteById(id);       
        return "redirect:/stores";
    }
    
}
